server/sonar-server/src/main/java/org/sonar/server/user/index/UserIndexDefinition.java

```diff
@@ -56,39 +55,14 @@ public void define(IndexDefinitionContext context) {
 
     index.configureShards(settings, 1);
 
-    index.getSettings()
-      // NGram filter (not edge) for logins and names
-      .put("index.analysis.filter.ngram_filter.type", "nGram")
-      .put("index.analysis.filter.ngram_filter.min_gram", 2)
-      .put("index.analysis.filter.ngram_filter.max_gram", 15)
-      .putArray("index.analysis.filter.ngram_filter.token_chars", "letter", "digit", "punctuation", "symbol")
-
-      // NGram index analyzer
-      .put("index.analysis.analyzer.index_ngrams.type", "custom")
-      .put("index.analysis.analyzer.index_ngrams.tokenizer", "whitespace")
-      .putArray("index.analysis.analyzer.index_ngrams.filter", "trim", "lowercase", "ngram_filter")
-
-      // NGram search analyzer
-      .put("index.analysis.analyzer.search_ngrams.type", "custom")
-      .put("index.analysis.analyzer.search_ngrams.tokenizer", "whitespace")
-      .putArray("index.analysis.analyzer.search_ngrams.filter", "trim", "lowercase");
-
     // type "user"
     NewIndex.NewIndexType mapping = index.createType(TYPE_USER);
-    mapping.stringFieldBuilder(FIELD_LOGIN).addSubField(SEARCH_SUB_SUFFIX, buildGramSearchField()).build();
-    mapping.stringFieldBuilder(FIELD_NAME).addSubField(SEARCH_SUB_SUFFIX, buildGramSearchField()).build();
-    mapping.stringFieldBuilder(FIELD_EMAIL).addSubField(SEARCH_SUB_SUFFIX, buildGramSearchField()).enableSorting().build();
+    mapping.stringFieldBuilder(FIELD_LOGIN).enable(USER_SEARCH_GRAMS_ANALYZER).build();
```

Comment:
The fact that a sub-field is created does not appear anymore with this method, whereas it's important when using it (see `USER_SEARCH_GRAMS_ANALYZER.subField(FIELD_LOGIN)` in `UserIndex`).